packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -428,11 +454,28 @@ module.exports = {
       const attributes = Object.keys(model.attributes)
         .reduce((acc, attribute) => {
           // Convert our type to the GraphQL type.
-          acc[attribute] = this.convertType(model.attributes[attribute]);
+          acc[attribute] = this.convertType({
+            definition: model.attributes[attribute],
+            modelName: globalId,
+            attributeName: attribute,
+          });
 
           return acc;
         }, initialState);
 
+      // Detect enum and generate it for the schema definition
+      const enums = Object.keys(model.attributes)
+        .reduce((acc, attribute) => {
+          const definition = model.attributes[attribute];
+
+          if (definition.type && definition.type === 'enumeration') {
```

Comment:
It could be better to use the `filter` method right after the `Object.keys(model.attributes)` instead.

```js
const enums = Object.keys(model.attributes)
  .filter(definition => definition.type === 'enumeration')
  .reduce((acc, attribute) => {
    ....
   }, []).join(' ');
```

It's a very little improvement but I tried to keep the codebase as consistent as possible 👍

Reply:
Ok I see, I also use to replace a `.reduce()` by `.filter().map()`

Reply:
The second one is shorter because you have fewer entries to loop on but I understand. It doesn't have any real performances impact on a small array like that and I think it's more understandable. Btw, you can merge the PR, though ;)